src/terminal/Grid.h

```diff
@@ -199,7 +252,7 @@ struct LogicalLines
                 return *this;
             }
 
-            Require(!lines.get()[unbox<int>(next)].wrapped());
+            // Require(!lines.get()[unbox<int>(next)].wrapped());
```

Comment:
This is a "logical line iterator" so it should give access to logical lines in an atomic way, e.g. spanning multiple lines. So I think `wrapped` flag must be always unset. Why did you comment this out?

Reply:
I also added a function `logicalLinesFrom(LineOffset offset)` which is supposed to give you all the logical lines from offset to ` pageSize_.lines - 1` which mean you can specify offset to be a line number in the middle of a group of wrapped lines, if that happens then when you increment the iterator this precondition will fail.